webrtc/videoroom/assets/src/membraneWebRTC.ts

```diff
@@ -129,26 +125,52 @@ export class MembraneWebRTC {
     this.channel.on("replaceTrack", (data: any) => {
       const oldTrackId = data.data.oldTrackId;
       const newTrackId = data.data.newTrackId;
-      const newStream = this.midToStream.get(newTrackId)!;
       const oldStream = this.midToStream.get(oldTrackId)!;
-      const participant = this.participants.find(({ mids }) =>
-        mids.includes(newTrackId)
+      const newStream = this.midToStream.get(newTrackId)!;
+      const oldTrack = oldStream.getTrackById(oldTrackId)!;
+      const newTrack = newStream.getTrackById(newTrackId)!;
+      const oldParticipant = this.participants.find(({ mids }) =>
+        mids.includes(oldTrackId)
       );
-
-      this.callbacks.onReplaceStream?.(
-        oldStream,
-        newStream,
-        participant?.displayName ?? ""
+      const newParticipant = this.participants.find(({ mids }) =>
+        mids.includes(newTrackId)
       );
+      const oldCtx = {
+        track: oldTrack,
+        stream: oldStream,
+        label: oldParticipant?.displayName ?? "",
+        isScreenSharing:
+          oldParticipant?.mids
+            .find((mid) => mid === oldTrackId)!
```

Comment:
I'm a bit confused right now. Is `mid` the same as `trackId`? I believe that the browser itself sets the track id and what we actually send from backend is just a `mid`. If so, then doing the `oldStream.getTrackById` is just wrong as there will probably be no such track (mid != id).

Besides that, you don't have to use bang operator here:
```
oldParticipants?.mids.find((mid) => mid === oldTrackId)?.includes("SCREEN") || false
```
This should be enough but again I don't know if we are using the `mid` and `trackId` correctly.

Reply:
`oldTrackId` is `mid` sent by our server so `isScreenSharing` will be set correctly. 

Regaring `oldStream.getTrackById` you are right - this won't work on Firefox but it should work on Chrome as Track.id on chrome is its mid. I forgot about this

Reply:
I don't know about the last thing with chrome setting `trackId` to its `mid`. I specifically remember that I used mids instead of track ids on chrome due to it being auto generated where mid was the same as on the server side.

Reply:
You are of course right. This won't work neither in Chrome nor in Firefox :man_facepalming: